pkg/clusterapi/identity_test.go

```diff
@@ -0,0 +1,403 @@
+package clusterapi_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	bootstrapv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1beta1"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/clusterapi"
+)
+
+func TestAwsIAMAuth(t *testing.T) {
```

Comment:
nit: can we name these tests following the tested method:
```suggestion
func TestIdentityAuthForAwsIAMAuth(t *testing.T) {
```